Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -100,26 +100,51 @@ class TIFFLoadingContext {
         }
     }
 
+    Optional<u8> alpha_channel_index() const
+    {
+        if (m_metadata.extra_samples().has_value()) {
+            auto const extra_samples = m_metadata.extra_samples().value();
+            for (u8 i = 0; i < extra_samples.size(); ++i) {
```

Comment:
(note for thakis: I've re-read the spec carefully and actually all extra samples must be after normal channels, but nothing specific is said about the alpha channel.)